common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java

```diff
@@ -149,16 +149,23 @@ public void registerExecutor(
       ExecutorShuffleInfo executorInfo) {
     AppExecId fullId = new AppExecId(appId, execId);
     logger.info("Registered executor {} with {}", fullId, executorInfo);
-    try {
-      if (db != null) {
-        byte[] key = dbAppExecKey(fullId);
-        byte[] value = mapper.writeValueAsString(executorInfo).getBytes(StandardCharsets.UTF_8);
-        db.put(key, value);
+    if ("org.apache.spark.shuffle.sort.SortShuffleManager".equals(executorInfo.shuffleManager)
```

Comment:
I'd reorganize this a bit to avoid some repetition and indentation. e.g.:

```
List<String> knownManagers = Arrays.asList("org.apache.spark.shuffle.sort.SortShuffleManager",
  "org.apache.spark.shuffle.unsafe.UnsafeShuffleManager")

if (!knownManagers.contains(executorInfo.shuffleManager)) {
   throw SomeException()
}

// Register executor.
```